### PR TITLE
feat: add profile-inherit command

### PR DESCRIPTION
**Pull request by jpower432**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [X] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [X] My PR meets testing requirements.
- [X] All new and existing tests passed.
- [X] All commits are signed-off.

## Summary

This PR adds a new subcommand for `author` called `profile-inherit`. This initializes and seeds a profile with information from a leveraged SSP and parent parent. The `with-ids` field is populate with a control delta from the profile and inherited controls.

Closes https://github.com/IBM/compliance-trestle/issues/1388

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

**Comment by jpower432** [comment]
cc @afflom

File: tests/trestle/core/commands/author/profile_test.py [hunk 1]
```diff
@@ -1016,3 +1016,102 @@ def test_profile_resolve_failures(tmp_trestle_dir: pathlib.Path, monkeypatch: Mo
...
+        prof.Profile,
+        FileContentType.JSON
+    )
+
+    assert result_prof.imports[0].href == 'trestle://profiles/simple_test_profile/profile.json'
+    assert len(result_prof.imports[0].include_controls[0].with_ids) == 2
+    assert len(result_prof.imports[0].exclude_controls[0].with_ids) == 1
+    assert result_prof.imports[0].exclude_controls[0].with_ids[0] == excluded
+
+    # Test with a profile that has more controls that the ssp
```
Comment (AleJo2995) [inline tests/trestle/core/commands/author/profile_test.py:1043]: than* typo

File: tests/trestle/core/commands/author/profile_test.py [hunk 2]
```diff
@@ -1016,3 +1016,102 @@ def test_profile_resolve_failures(tmp_trestle_dir: pathlib.Path, monkeypatch: Mo
...
+        prof.Profile,
+        FileContentType.JSON
+    )
+
+    assert result_prof.imports[0].href == 'trestle://profiles/simple_test_profile_more/profile.json'
+    assert len(result_prof.imports[0].include_controls[0].with_ids) == 3
+    assert len(result_prof.imports[0].exclude_controls[0].with_ids) == 1
+    assert result_prof.imports[0].exclude_controls[0].with_ids[0] == excluded
+
+    # Test with a profile that has less controls that the ssp
```
Comment (AleJo2995) [inline tests/trestle/core/commands/author/profile_test.py:1060]: than* typo

File: trestle/core/commands/author/profile.py [hunk 3]
```diff
@@ -536,3 +539,201 @@ def resolve_profile(
...
+
+        Returns:
+            None
+        """
+        exclude_with_ids: Set[prof.withId] = set()
+        components_by_id: Dict(str, List[ssp.ByComponent]) = {}
+
+        # Create dictionary containing all by-components by control for faster searching
+        for implemented_requirement in leveraged_ssp.control_implementation.implemented_requirements:
+            components: List[ssp.ByComponent] = []
```
Comment (AleJo2995) [inline trestle/core/commands/author/profile.py:617]: Pls change variable name to by_components to be more meaningful. I agree that in the end are components, but for readability purposes I´d prefer to call this variable as it is

File: trestle/core/commands/author/profile.py [hunk 4]
```diff
@@ -536,3 +539,201 @@ def resolve_profile(
...
+
+        # Looping by controls in the catalog because the ids in the profile should
+        # be a subset of the catalog and not the ssp controls.
+        catalog_control_ids: Set[str] = set(catalog_api._catalog_interface.get_control_ids())
+        for control_id in catalog_control_ids:
+
+            if control_id not in components_by_id:
+                continue
+
+            by_comps = components_by_id[control_id]
```
Comment (AleJo2995) [inline trestle/core/commands/author/profile.py:635]: where´s the by_comps declaration? Am I missing it somewhere?

File: trestle/core/commands/author/profile.py [hunk 5]
```diff
@@ -536,3 +539,201 @@ def resolve_profile(
...
+            if by_comps is not None and ProfileInherit._is_inherited(by_comps):
+                exclude_with_ids.add(control_id)
+
+        include_with_ids: Set[prof.withId] = catalog_control_ids - exclude_with_ids
+
+        include_controls = [prof.SelectControlById(with_ids=sorted(include_with_ids))]
```
Comment (AleJo2995) [inline trestle/core/commands/author/profile.py:641]: where´s include_controls declaration? As we try to keep strongly type variables, please declare variables with its type when possible :)

File: trestle/core/commands/author/profile.py [hunk 6]
```diff
@@ -536,3 +539,201 @@ def resolve_profile(
...
+        exclude_controls = [prof.SelectControlById(with_ids=sorted(exclude_with_ids))]
```
Comment (AleJo2995) [inline trestle/core/commands/author/profile.py:642]: where´s exclude_controls declaration? As we try to keep strongly type variables, please declare variables with its type when possible :)

**Comment by AleJo2995** [comment]
@jpower432 I´ve added some feedback regarding style than functionality. I have tested the functionality and all seems correct to me. I´ll make a second pass today to it and provide more feedback if needed. If not, we´re all set :)

File: trestle/core/commands/author/profile.py (hunk 5 above)
Comment (jpower432) [reply trestle/core/commands/author/profile.py]: Thanks @AleJo2995! Ended up removing the intermediary variable and just set the value on the import. Please let me know if this adequately addresses your feedback.

File: trestle/core/commands/author/profile.py (hunk 6 above)
Comment (jpower432) [reply trestle/core/commands/author/profile.py]: Same.

**Review by AleJo2995 — APPROVED** [review]
LGTM

**Pull request merged**